### PR TITLE
Issue #2572119: Replace evaluate() with doEvaluate() on EntityHasFiel…

### PR DESCRIPTION
**Pull request by emclaughlin1215**
…d condition.

See d.o issue: https://www.drupal.org/node/2572119

### CODE REVIEW

File: src/Plugin/Condition/EntityHasField.php [hunk 1]
```diff
@@ -33,11 +34,17 @@
 class EntityHasField extends RulesConditionBase {
 
   /**
-   * {@inheritdoc}
+   * Checks if a given entity has a given field.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
```
Comment (klausi) [inline src/Plugin/Condition/EntityHasField.php]: Type hint should be FieldableEntityInterface instead.

File: src/Plugin/Condition/EntityHasField.php [hunk 2]
```diff
@@ -33,11 +34,17 @@
...
+   * @param string $field
+   *   The field to check for on the entity
+   *
+   * @return bool
+   *   TRUE if the provided entity has the provided field.
    */
-  public function evaluate() {
-    $entity = $this->getContextValue('entity');
-    $field = $this->getContextValue('field');
+  protected function doEvaluate(EntityInterface $entity, $field) {
```
Comment (klausi) [inline src/Plugin/Condition/EntityHasField.php]: Same here, we can only call hasField() on FieldableEntityInterface entities.

**Comment by emclaughlin1215** [comment]
Addressed the comments and pushed up changes.

File: src/Plugin/Condition/EntityHasField.php [hunk 3]
```diff
@@ -33,11 +34,17 @@
 class EntityHasField extends RulesConditionBase {
 
   /**
-   * {@inheritdoc}
+   * Checks if a given entity has a given field.
+   *
+   * @param \Drupal\Core\Entity\FieldableEntityInterface $entity
+   *   The entity to check for the provided field
```
Comment (klausi) [inline src/Plugin/Condition/EntityHasField.php]: missing dot at the end.

File: src/Plugin/Condition/EntityHasField.php [hunk 4]
```diff
@@ -33,11 +34,17 @@
...
+   * @param string $field
+   *   The field to check for on the entity
```
Comment (klausi) [inline src/Plugin/Condition/EntityHasField.php]: missing dot at the end.

**Comment by emclaughlin1215** [comment]
Addressed comments and pushed up changes.

**Comment by klausi** [comment]
merged, thanks!

**Pull request closed without merging**